test/functional/test-viewer.js

```diff
@@ -1096,6 +1096,34 @@ describe('Viewer', () => {
   });
 
   describe('referrer', () => {
+    function test(referrer, toBeTrusted) {
+      it('testing ' + referrer, () => {
+        const viewer = new Viewer(ampdoc);
+        expect(viewer.isTrustedReferrer_(referrer)).to.equal(toBeTrusted);
+      });
+    }
+
+    describe('should not trust host as referrer with http', () => {
+      test('http://t.co/asdf', false);
+    });
+
+    describe('should trust whitelisted hosts', () => {
+      test('https://t.co/asdf', true);
+    });
+
+    describe('should not trust non-whitelisted hosts', () => {
+      test('https://www.t.co/asdf', false);
+      test('https://t.com/asdf', false);
+      test('https://t.cn/asdf', false);
+    });
+
+    it('should return true for whitelisted hosts', () => {
```

Comment:
nit: could you update the description here to indicate `isTrustedReferrer` function is tested here.

Reply:
updated